common/test/common_tests/logic/swap_and_reset_test.cljc

```diff
@@ -0,0 +1,175 @@
+                                                      page'
+                                                      file'
+                                                      libraries')
+                                  (ctk/build-swap-slot-group))]
+
+    ;; Check
+    (t/is (not= (:component-id copy) (:component-id swapped)))
+    (t/is (= (:id component-2) (:component-id swapped)))
+    (t/is (= (:id file) (:component-file swapped)))
+    (t/is (contains? (:touched swapped) slot))))
+
+(t/deftest test-swap-and-reset-override
+  (let [;; Setup
+        file
+        (-> (thf/sample-file :file1)
+            (thc/add-simple-component :component-1 :component-1-main-root :component-1-main-child)
+            (thc/add-frame :component-container)
+            (thf/instantiate-component :component-1 :component-1-copy-root {:parent-label :component-container})
+            (thf/make-component :component-container-main :component-container)
+            (thf/instantiate-component :component-container-main :component-container-instance)
+            (thc/add-simple-component :component-2 :component-2-main-root :component-2-main-child))
+
+        page        (thf/current-page file)
+        component-1 (thf/get-component file :component-1)
+        component-2 (thf/get-component file :component-2)
+
+        copy
+        (->>
+         (thf/get-shape file :component-container-instance)
+         :shapes
+         first
+         (thf/get-shape-by-id file))
+
+        ;; Action
+        [new-shape all-parents changes-swap]
+        (cflh/generate-component-swap (pcb/empty-changes)
+                                      (:objects page)
+                                      copy
+                                      (:data file)
+                                      page
+                                      {(:id  file) file}
+                                      (:id component-2)
+                                      0
+                                      nil
+                                      {})
+
+        file-swap (thf/apply-changes file changes-swap)
+        page-swap (thf/current-page file-swap)
+
+        changes
+        (cflh/generate-reset-component (pcb/empty-changes)
+                                       file-swap
+                                       {(:id  file-swap) file-swap}
+                                       page-swap
+                                       (:id new-shape)
+                                       true)
+
+        file' (thf/apply-changes file changes)
+        page' (thf/current-page file')
+        ;; Get
+        reset
+        (->>
+         (thf/get-shape file' :component-container-instance)
+         :shapes
+         first
+         (thf/get-shape-by-id file'))
+
+        component-1-copy-root (thf/get-shape file' :component-1-copy-root)]
+
+    ;; Check
+    (t/is (= (:id component-1) (:component-id reset)))
+    (t/is (= nil (:touched reset)))))
```

Comment:
It in the future there is something touched it might break. I'll check specifically for the slot:

`(t/is (nil? (ctk/get-swap-slot reset)))`